### PR TITLE
Add version validation for IntelliJ plugins on release

### PR DESCRIPTION
**Pull request by buildbreaker**
To locally run:
```
./gradlew runPluginVerifier
```

I got the versions from this script: https://github.com/bufbuild/intellij-buf/pull/33

Just a note: IJ marketplace does ensure that downloaded plugins work with the user's IDE. They use the same validation strategy as the plugin used. I got my information from: https://plugins.jetbrains.com/docs/intellij/build-number-ranges.html.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -111,4 +111,8 @@ tasks {
         // https://plugins.jetbrains.com/docs/intellij/deployment.html#specifying-a-release-channel
         channels.set(listOf(properties("pluginVersion").split('-').getOrElse(1) { "default" }.split('.').first()))
     }
+
+    runPluginVerifier {
+        ideVersions.set(listOf("IIC-222.3739.24", "IIC-221.6008.8"))
```
Comment (pkwarren) [inline build.gradle.kts:116]: Where are we getting these from? It might be good to add a comment for the next time we might need to update.

Comment (buildbreaker) [reply build.gradle.kts:116]: I used this: https://github.com/bufbuild/intellij-buf/pull/33.

I can add it

File: build_versions.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        # IIC is IntelliJ IDEA Community Edition
+        if "code" in result and result["code"] in ["IIC"]:
+            for release in result["releases"]:
+                if "build" not in release:
+                    continue
+                build = release["build"]
+                split = build.split(".")
+                if len(split) < 1:
+                    continue
+                if int(split[0]) >= 220:
```
Comment (amckinney) [inline build_versions.py:40]: It'd also be good to add a note for this magic number.

**Review by amckinney — APPROVED** [review]
Looks good as long as we add a couple more comments for posterity.

**Review by pkwarren — APPROVED** [review]

**Pull request merged**